browser-extension/mv3/src/utility-scripts/cacheRules.js

```diff
@@ -0,0 +1,15 @@
+const params = JSON.parse(document.currentScript.dataset.params);
+const { rules, ruleType } = params;
+let rulesJSON = null;
+try {
+  rulesJSON = JSON.parse(rules);
+} catch (e) {
+  //do nothing
+}
+
+if (rulesJSON && ruleType) {
+  const PUBLIC_NAMESPACE = "__REQUESTLY__";
+
+  window[PUBLIC_NAMESPACE] = window[PUBLIC_NAMESPACE] || {};
+  window[PUBLIC_NAMESPACE][`${ruleType}Rules`] = rulesJSON;
+}
```

Comment:
We can generalise this. Instead of just doing it for rules, we can cache anything JSON that is passed to the script in dataset.

Something like this
```
const params = JSON.parse(document.currentScript.dataset.params);
const PUBLIC_NAMESPACE = "__REQUESTLY__";
Object.entries((key, value)=> {
    window[PUBLIC_NAMESPACE] = window[PUBLIC_NAMESPACE] || {};
    window[PUBLIC_NAMESPACE][`${key}`] = JSON.parse(value);
})
}
```